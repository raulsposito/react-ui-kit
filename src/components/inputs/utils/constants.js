export const BASIC_INPUT_TYPES = {
  TEXT: 'text',
  EMAIL: 'email',
  TEL: 'tel',
  PASSWORD: 'password',
  NUMBER: 'number',
  NAME: 'name',
  TEXTAREA: 'textarea'
}

export const INPUT_TYPES = {
  ...BASIC_INPUT_TYPES,
  DATE: 'date'
}
