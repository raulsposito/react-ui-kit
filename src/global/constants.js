export const BREAKPOINTS = {
  MOBILE: 576,
  TABLET: 875,
  DESKTOP: 992
}
