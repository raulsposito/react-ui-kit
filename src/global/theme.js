// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
