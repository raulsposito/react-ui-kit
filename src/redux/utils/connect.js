import { useInjectReducer, useInjectSaga } from 'redux-injectors'

export default ({ key, saga, reducer }) => {
  useInjectSaga({ key, saga })
  useInjectReducer({ key, reducer })
}
