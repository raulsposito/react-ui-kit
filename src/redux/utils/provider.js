import React from 'react'
import PropTypes from 'prop-types'
import { Provider } from 'react-redux'
import { configureStore } from '@redux/store'

const Pr = ({ children, initialState, key }) => {
  const init = key ? { [key]: { ...initialState } } : initialState
  const store = configureStore(init)

  return (
    <Provider store={store}>
      {children}
    </Provider>
  )
}

Pr.propTypes = {
  key: PropTypes.string,
  children: PropTypes.node,
  initialState: PropTypes.object
}

export default Pr
