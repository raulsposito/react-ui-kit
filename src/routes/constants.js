export const NAV_WIDTH = '120px'
export const HEADER_HEIGHT = '100px'
