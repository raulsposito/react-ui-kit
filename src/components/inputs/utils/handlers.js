import { INPUT_TYPES } from './constants'

export const handleTextColorState = (error, focused, color) => {
  if (error) return 'error'
  if (focused) return color || 'primary'
  return 'contrast'
}

export const handleTypes = type => {
  switch (type) {
    case INPUT_TYPES.TEXT: return 'text'
    case INPUT_TYPES.NAME: return 'text'
    case INPUT_TYPES.TEL: return 'tel'
    case INPUT_TYPES.EMAIL: return 'email'
    case INPUT_TYPES.NUMBER: return 'number'
    case INPUT_TYPES.PASSWORD: return 'password'
    default: return 'text'
  }
}
