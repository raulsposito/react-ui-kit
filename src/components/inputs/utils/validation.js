import { INPUT_TYPES } from '@components/inputs/utils/constants'

export const validatePassword = password => {
  const regex = /^(?=.*\d)(?=.*[a-zA-Z]).{8,}$/g
  const valid = (password) ? !!password.trim().match(regex) : true
  const returnObj = {
    isValid: valid,
    message: (!valid) ? 'Is not a valid password' : null
  }

  return returnObj
}

export const validateNumber = number => {
  const valid = number ? !Number.isNaN(number) : true
  const returnObj = {
    isValid: valid,
    message: (!valid) ? 'Is not a valid number' : null
  }

  return returnObj
}

export const validateName = name => {
  const regex = /^[a-zA-ZÀ-ÿ\u00f1\u00d1]+(\s*[a-zA-ZÀ-ÿ\u00f1\u00d1]*)*[a-zA-ZÀ-ÿ\u00f1\u00d1]+$/g
  const valid = (name) ? !!name.trim().match(regex) : true
  const returnObj = {
    isValid: valid,
    message: (!valid) ? 'Is not a valid name' : null
  }

  return returnObj
}

export const validateEmail = email => {
  // eslint-disable-next-line max-len
  const regex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  const valid = (email) ? !!email.match(regex) : true
  const returnObj = {
    isValid: valid,
    message: (!valid) ? 'Is not a valid email' : null
  }

  return returnObj
}

export const validateTextArea = ({ value, maxLength, required }) => {
  if (!value && required) return { isValid: false, message: 'Required Field' }

  const valid = !(value.length === maxLength)

  const returnObj = {
    isValid: valid,
    message: (!valid) ? 'Is not a valid textarea' : null
  }

  return returnObj
}

export const validateField = ({ type, value, required }) => {
  if (!value && required) return { isValid: false, message: 'Required Field' }

  switch (type) {
    case INPUT_TYPES.TEXT:
      return { isValid: true, message: null }
    case INPUT_TYPES.EMAIL:
      return validateEmail(value)
    case INPUT_TYPES.TEL:
    case INPUT_TYPES.DATE:
    case INPUT_TYPES.PASSWORD:
      return validatePassword(value)
    case INPUT_TYPES.NUMBER:
      return validateNumber(value)
    case INPUT_TYPES.NAME:
      return validateName(value)
    default:
      return { isValid: true, message: null }
  }
}
