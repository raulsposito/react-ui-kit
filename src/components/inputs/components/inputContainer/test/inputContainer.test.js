import React from 'react'
import { act, fireEvent, render, screen } from '@testing-library/react'

import { colors } from '@global/theme'
import { INPUT_TYPES } from '@components/inputs/utils/constants'

import InputAdornment from '../../inputAdornment'

import InputContainer from '..'

const adornmentData = {
  startIcon: 'visibility',
  endIcon: 'help',
  color: 'system',
  onClick: jest.fn()
}

const startAdornment = (
  <InputAdornment onClick={adornmentData.onClick} icon={adornmentData.startIcon} color={adornmentData.color} />
)
const endAdornment = (
  <InputAdornment onClick={adornmentData.onClick} icon={adornmentData.endIcon} color={adornmentData.color} />
)

const mockData = {
  name: 'inputContainer',
  placeholder: 'input placeholder',
  value: 'controlled value',
  error: 'Error Text',
  startAdornment,
  endAdornment,
  onChange: jest.fn(),
  onFocus: jest.fn(),
  onBlur: jest.fn()
}

describe('<InputContainer />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe('Disabled', () => {
  it('is disabled', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} disabled />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toBeDisabled()
  })
})

describe('Required', () => {
  it('is required', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} required />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toBeRequired()
  })
})

describe('Placeholder', () => {
  it('input has placeholder', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toBeInTheDocument()
  })
})

describe('Name', () => {
  it('input has name', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder, { name: mockData.name })
    expect(input).toBeInTheDocument()
  })
})

describe('Input event', () => {
  it('calls function on change', () => {
    render(
      <InputContainer
        type={INPUT_TYPES.TEXT}
        name={mockData.name}
        placeholder={mockData.placeholder}
        onChange={mockData.onChange}
      />
    )

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    fireEvent.change(input, { target: { value: 'new value' } })
    expect(mockData.onChange).toHaveBeenCalledTimes(1)
  })

  it('calls function on focus', () => {
    render(
      <InputContainer
        type={INPUT_TYPES.TEXT}
        name={mockData.name}
        placeholder={mockData.placeholder}
        onFocus={mockData.onFocus}
      />
    )

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    fireEvent.focus(input)
    expect(mockData.onFocus).toHaveBeenCalledTimes(1)
  })

  it('calls function on blur', () => {
    render(
      <InputContainer
        type={INPUT_TYPES.TEXT}
        name={mockData.name}
        placeholder={mockData.placeholder}
        onBlur={mockData.onBlur}
      />
    )

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    fireEvent.blur(input)
    expect(mockData.onBlur).toHaveBeenCalledTimes(1)
  })
})

describe('Input types', () => {
  it('has not selected type', () => {
    render(<InputContainer name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder, { type: INPUT_TYPES.TEXT })
    expect(input).toBeInTheDocument()
  })

  it('text', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'text')
  })

  it('email', () => {
    render(<InputContainer type={INPUT_TYPES.EMAIL} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'email')
  })

  it('telephone', () => {
    render(<InputContainer type={INPUT_TYPES.TEL} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'tel')
  })

  it('number', () => {
    render(<InputContainer type={INPUT_TYPES.NUMBER} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'number')
  })

  it('password', () => {
    render(<InputContainer type={INPUT_TYPES.PASSWORD} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'password')
  })

  it('name', () => {
    render(<InputContainer type={INPUT_TYPES.NAME} name={mockData.name} placeholder={mockData.placeholder} />)

    const input = screen.queryByPlaceholderText(mockData.placeholder)
    expect(input).toHaveAttribute('type', 'text')
  })
})

describe('Handle State Color', () => {
  describe('Handle one state at time', () => {
    it('has not state changes', () => {
      render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
    })

    it('is disabled', () => {
      render(
        <InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} disabled />
      )

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.disabled}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
    })

    it('has error', () => {
      render(
        <InputContainer
          type={INPUT_TYPES.TEXT}
          name={mockData.name}
          placeholder={mockData.placeholder}
          error={mockData.error}
        />
      )

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
    })

    it('is focused', () => {
      render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

      const input = screen.queryByPlaceholderText(mockData.placeholder)
      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      act(() => input.focus())

      expect(input).toHaveFocus()
      // :focus:within pseudoclass used to color the border of external div of input can't be tested
      // expect(inputContainer).toHaveStyle({ borderColor: `${colors.primary.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.regular}` })
    })
  })

  describe('Handle multiple states at time', () => {
    it('is disabled and has error', () => {
      render(
        <InputContainer
          type={INPUT_TYPES.TEXT}
          name={mockData.name}
          placeholder={mockData.placeholder}
          disabled
          error={mockData.error}
        />
      )

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
    })

    it('is disabled and focused', () => {
      render(
        <InputContainer
          type={INPUT_TYPES.TEXT}
          name={mockData.name}
          placeholder={mockData.placeholder}
          disabled
        />
      )

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      const input = screen.queryByPlaceholderText(mockData.placeholder)
      act(() => input.focus())

      expect(inputContainer).toHaveStyle({ borderColor: `${colors.system.disabled}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
    })

    it('is disabled, focused and has error', () => {
      render(<InputContainer name={mockData.name} placeholder={mockData.placeholder} disabled error={mockData.error} />)

      const inputContainer = screen.queryByPlaceholderText(mockData.placeholder).closest('div')
      const input = screen.queryByPlaceholderText(mockData.placeholder)
      act(() => input.focus())

      expect(inputContainer).toHaveStyle({ borderColor: `${colors.error.regular}` })
      expect(inputContainer).toHaveStyle({ backgroundColor: `${colors.background.disabled}` })
    })
  })
})

describe('Adornments', () => {
  it('has no adornments', () => {
    render(<InputContainer type={INPUT_TYPES.TEXT} name={mockData.name} placeholder={mockData.placeholder} />)

    const startAdornmentDiv = screen.queryByText('visibility')?.closest('div')
    const endAdornmentDiv = screen.queryByText('help')?.closest('div')

    expect(startAdornmentDiv).toBeUndefined()
    expect(endAdornmentDiv).toBeUndefined()
  })

  it('has start adornment only', () => {
    render(
      <InputContainer
        name={mockData.name}
        placeholder={mockData.placeholder}
        startAdornment={mockData.startAdornment}
      />
    )

    const startAdornmentDiv = screen.queryByText('visibility')?.closest('div')
    const endAdornmentDiv = screen.queryByText('help')?.closest('div')

    expect(startAdornmentDiv).toBeInTheDocument()
    expect(endAdornmentDiv).toBeUndefined()
  })

  it('has end adornment only', () => {
    render(
      <InputContainer
        name={mockData.name}
        placeholder={mockData.placeholder}
        endAdornment={mockData.endAdornment}
      />
    )

    const startAdornmentDiv = screen.queryByText('visibility')?.closest('div')
    const endAdornmentDiv = screen.queryByText('help')?.closest('div')

    expect(startAdornmentDiv).toBeUndefined()
    expect(endAdornmentDiv).toBeInTheDocument()
  })

  it('has start and end adornments at the same time', () => {
    render(
      <InputContainer
        name={mockData.name}
        startAdornment={mockData.startAdornment}
        endAdornment={mockData.endAdornment}
      />
    )

    const startAdornmentDiv = screen.queryByText('visibility')?.closest('div')
    const endAdornmentDiv = screen.queryByText('help')?.closest('div')

    expect(startAdornmentDiv).toBeInTheDocument()
    expect(endAdornmentDiv).toBeInTheDocument()
  })
})
