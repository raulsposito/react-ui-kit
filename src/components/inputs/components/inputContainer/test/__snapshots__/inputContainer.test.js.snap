// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputContainer /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jrAFXE ilTBPT"
  >
    <input
      class="sc-pGacB fcysIK"
      name="inputContainer"
      placeholder="input placeholder"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
